### PR TITLE
COTR-82 Spawn infinite enemies with increasing difficulty

### PR DESCRIPTION
**Pull request by Arnelys**
Spawner now picks a random enemy to spawn each time. Also every 10 seconds it decreases timer wait time by 0.1 seconds so that enemies spawn faster over time.

### CODE REVIEW

File: Scenes/Utility/EnemySpawner.cs [hunk 1]
```diff
@@ -46,26 +64,18 @@ private Vector2 GetRandomPosition()
...
-                }
-            }
+            time = 0;
+            timer.WaitTime -= 0.1;
         }
+
+        var random = new Random();
+        int spawn = random.Next(0, Enemies.Count); //Generating random index of which enemy to spawn
+
+        var enemyScene = Enemies[spawn].Instantiate<Infantry>();
```
Comment (Elijuszek) [inline Scenes/Utility/EnemySpawner.cs:76]: Don't use var, the of this variable is Infantry

File: Scenes/Utility/EnemySpawner.cs [hunk 2]
```diff
@@ -46,26 +64,18 @@ private Vector2 GetRandomPosition()
...
-                        enemySpawn.GlobalPosition = GetRandomPosition();
-                        counter++;
-                    }
-                }
-            }
+            time = 0;
+            timer.WaitTime -= 0.1;
         }
+
+        var random = new Random();
```
Comment (Elijuszek) [inline Scenes/Utility/EnemySpawner.cs:73]: Don't use var, the type of this variable is Random

**Review by Elijuszek — CHANGES_REQUESTED** [review]

File: Scenes/Utility/EnemySpawner.cs (hunk 1 above)
Comment (Arnelys) [reply Scenes/Utility/EnemySpawner.cs]: Done

File: Scenes/Utility/EnemySpawner.cs (hunk 2 above)
Comment (Arnelys) [reply Scenes/Utility/EnemySpawner.cs]: Done

**Review by Elijuszek — APPROVED** [review]
LGTM!

**Pull request merged**